exercises/clock/test/test_clock.c

```diff
@@ -0,0 +1,489 @@
+}
+
+void test_subtract_minutes(void)
+{
+   int hour = 10;
+   int minute = 3;
+   int minute_offset = -3;
+   char *expected_clock_text = "10:00";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_to_previous_hour(void)
+{
+   int hour = 10;
+   int minute = 3;
+   int minute_offset = -30;
+   char *expected_clock_text = "09:33";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_an_hour(void)
+{
+   int hour = 10;
+   int minute = 3;
+   int minute_offset = -70;
+   char *expected_clock_text = "08:53";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_across_midnight(void)
+{
+   int hour = 0;
+   int minute = 3;
+   int minute_offset = -4;
+   char *expected_clock_text = "23:59";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_two_hours(void)
+{
+   int hour = 0;
+   int minute = 0;
+   int minute_offset = -160;
+   char *expected_clock_text = "21:20";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_two_hours_with_borrow(void)
+{
+   int hour = 6;
+   int minute = 15;
+   int minute_offset = -160;
+   char *expected_clock_text = "03:35";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_one_day(void)
+{
+   int hour = 5;
+   int minute = 32;
+   int minute_offset = -1500;
+   char *expected_clock_text = "04:32";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_two_days(void)
+{
+   int hour = 2;
+   int minute = 20;
+   int minute_offset = -3000;
+   char *expected_clock_text = "00:20";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+// these tests check equality.
+// Part of standard test collection
+// Included even though they don't make much sense in C
+
+// trying it with XMACROS !
+#define EXPAND_EQUALITY_TEST_AS_TEST(test_name, hour1, minute1, hour2, minute2, expected_equality)\
+   void test_name(void)\
+   {\
+   test_time_expected_equality(hour1, minute1, hour2, minute2, expected_equality);\
+   }\
+
+// below expands to multiple routines...that are all designed to test the equality of 2 different
+// clock representations to see if they are the same or not
+// that 2 clocks are equal...
+// the fields are:
+//    test name, time1_hour, time1_minute, time2_hour, time2_minute, (time1 == time2)
+
+#define EQUALITY_TESTS(ENTRY) \
```

Comment:
will comment them and see if CI passes...  then we can decide whether to leave them as tests or not...